framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -227,30 +264,31 @@ public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2
             }
             // If both are the same type, determine the type and merge:
             else if (AnnotationUtils.areSameIgnoringValues(a1, a2)) {
-                List<Object> a1Values =
-                        AnnotationUtils.getElementValueArray(a1, "value", Object.class, true);
-                List<Object> a2Values =
-                        AnnotationUtils.getElementValueArray(a2, "value", Object.class, true);
-                HashSet<Object> newValues = new HashSet<Object>(a1Values.size() + a2Values.size());
+                if (AnnotationUtils.areSameByClass(a1, IntRange.class)) {
+                    // special handling for IntRange
+                    Range range1 = getIntRange(a1);
+                    Range range2 = getIntRange(a2);
+                    return createIntRangeAnnotation(range1.union(range2));
+                } else {
+                    List<Object> a1Values =
```

Comment:
Hmm..these are the original codes that I didn't touch. Will try to find a cleaner implementation if possible